[FLINK-1766]Fix the bug of equals function of FSKey

matadorhong opened the pull request:
The equals function in org.apache.flink.core.fs.FileSystem.FSKey should first confirm whether obj == this, if obj is the same object.It should return true

StephanEwen commented:
Looks good. This seems more like an optimization, than a bug,

Before we can merge the change, it needs to be brought in sync with the code style. Currently, the checkstyle rules reject the code for using spaces rather than tabs.

StephanEwen commented:
If you are interested: Here are some simple guidelines on how to format code and create easy to merge commits for pull requests: 

http://flink.apache.org/coding_guidelines.html

matadorhong commented:
Great thanks to Ewen. I will first have a study of how to format code. Then pull a request.

matadorhong commented:
Thanks for Ewen. That help me a lot to understand how to contribute to Flink. I will follow the guide in the future.

StephanEwen commented:
No problem. We are happy that you are interested to contribute :-)

matadorhong commented:
@StephanEwen I have fixed the code style issue and changed the PR to an "Improvement" one in jira.
Finally, great thanks for your help. And I will take more time to contribute Flink.

hsaputra commented:
LGTM. Will merge later today. Thanks!
